app/models/auth_request.rb

```diff
@@ -14,4 +16,19 @@ def self.from_oauth_state(state)
 
     find_by(id: bits[1], oauth_state: bits[0])
   end
+
+  def redirect_path_is_safe
+    return if redirect_path.nil?
+    return if redirect_path.empty?
+
+    if redirect_path&.starts_with? "//"
+      errors.add(:redirect_path, "can't be protocol-relative")
+      return
+    end
+
+    return if redirect_path.starts_with? "/"
+    return if redirect_path.starts_with?("http://") && Rails.env.development?
```

Comment:
... I still think we should make dev work with https... but great thinking ahead fr the current setup!